src/EnergyPlus/General.cc

```diff
@@ -333,10 +513,23 @@ namespace General {
 			DY = Y0 - Y1;
 			if ( std::abs( DY ) < SMALL ) DY = SMALL;
 			// new estimation
-			XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			if ( ( HVACSystemRootFinding.TypeNum == Bisection ) ||
+				( HVACSystemRootFinding.TypeNum == Alternation && AltIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == RegulaFalsiThenBisection && NIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == BisectionThenRegulaFalsi && NIte <= HVACSystemRootFinding.NumOfIter ) ) {
+				if ( HVACSystemRootFinding.TypeNum == Alternation ) {
+					if ( AltIte >= 2 * HVACSystemRootFinding.NumOfIter ) AltIte = 0;
+				}
+				// Bisection
+				XTemp = ( X1 + X0 ) / 2.0;
+			} else {
+				// Regula Falsi
+				XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			}
```

Comment:
I wonder if a Select Case would be faster here. There would be 5 cases, but only 1 IF ELSE statement in three of them.